adapters/goal.ts

```diff
@@ -1,49 +1,60 @@
-import axios from 'axios'
-import * as goalAdapter from './goal.transformer'
-import { type Goal } from '~/interfaces/Goal'
-import { type PostGoal } from '~/interfaces/PostGoal'
-import { API } from '~/constants/api'
+import axios from "axios";
+import * as goalAdapter from "./goal.transformer";
+import { type Goal } from "~/interfaces/Goal";
+import { type PostGoal } from "~/interfaces/PostGoal";
+import { API } from "../constants/api";
 
 const goalSiteConfig = {
   headers: {
-    'Content-Type': 'application/vnd.api+json'
-  }
-}
+    "Content-Type": "application/vnd.api+json",
+  },
+};
 
-export const fetchGoals = async (): Promise<Goal []> => {
+export const fetchGoals = async (): Promise<Goal[]> => {
```

Comment:
this function is not handling errors gracefully. If the axios request fails, the catch() block will throw an error, which will be propagated to the caller of the fetchGoals()

Reply:
How should we handle this? Shouldn't caller of fetchGoals handle it?